x-pack/legacy/plugins/siem/public/components/open_timeline/open_timeline.tsx

```diff
@@ -31,56 +48,157 @@ export const OpenTimeline = React.memo<OpenTimelineProps>(
-            : ['duplicate', 'selectable']
+    const nTimelines = useMemo(
+      () => (
+        <FormattedMessage
+          id="xpack.siem.open.timeline.showingNTimelinesLabel"
+          defaultMessage="{totalSearchResultsCount} {totalSearchResultsCount, plural, one {timeline} other {timelines}} {with}"
+          values={{
+            totalSearchResultsCount,
+            with: (
+              <span data-test-subj="selectable-query-text">
+                {query.trim().length ? `${i18n.WITH} "${query.trim()}"` : ''}
+              </span>
+            ),
+          }}
+        />
+      ),
+      [totalSearchResultsCount, query]
+    );
+    const onCompleteActions = useCallback(
+      (closePopover: () => void) => {
+        closePopover();
+        if (tableRef != null && tableRef.current != null) {
+          tableRef.current.changeSelection([]);
         }
-        data-test-subj="timelines-table"
-        deleteTimelines={deleteTimelines}
-        defaultPageSize={defaultPageSize}
-        loading={isLoading}
-        itemIdToExpandedNotesRowMap={itemIdToExpandedNotesRowMap}
-        onOpenTimeline={onOpenTimeline}
-        onSelectionChange={onSelectionChange}
-        onTableChange={onTableChange}
-        onToggleShowNotes={onToggleShowNotes}
-        pageIndex={pageIndex}
-        pageSize={pageSize}
-        searchResults={searchResults}
-        showExtendedColumns={true}
-        sortDirection={sortDirection}
-        sortField={sortField}
-        totalSearchResultsCount={totalSearchResultsCount}
-      />
-    </EuiPanel>
-  )
+      },
+      [tableRef.current]
+    );
+
+    const getBatchItemsPopoverContent = useCallback(
+      (closePopover: () => void) => {
+        return (
+          <EuiContextMenuPanel
+            items={[
+              <EuiContextMenuItem key="ExportItemKey" disabled={selectedItems.length === 0}>
+                <TimelineDownloader
+                  selectedTimelines={selectedItems}
+                  onDownloadComplete={onCompleteActions.bind(null, closePopover)}
+                />
+              </EuiContextMenuItem>,
+              <EuiContextMenuItem key="DeleteItemKey" disabled={selectedItems.length === 0}>
+                <DeleteTimelineModalButton
+                  deleteTimelines={deleteTimelines}
+                  savedObjectIds={selectedItems?.reduce(
+                    (acc, item) =>
+                      item.savedObjectId != null ? [...acc, item.savedObjectId] : acc,
+                    [] as string[]
+                  )}
+                  title={
+                    selectedItems.length > 1
+                      ? i18n.SELECTED_TIMELINES(selectedItems.length)
+                      : `"${selectedItems[0]?.title}"`
+                  }
+                  onComplete={onCompleteActions.bind(null, closePopover)}
+                />
+              </EuiContextMenuItem>,
+            ]}
+          />
+        );
+      },
+      [selectedItems, dispatchToaster, history]
```

Comment:
I think that we need to add `onCompleteActions`,  `deleteTimelines` as a dependencies. I am not sure that we are using dispatchToaster and history in the callback.